compiler/rustc_typeck/src/check/fn_ctxt/mod.rs

```diff
@@ -0,0 +1,295 @@
+mod _impl;
+mod checks;
+mod suggestions;
+
+pub use _impl::*;
```

Comment:
`_impl` is a bit of an odd naming convention - I think you picked it because it has the main `impl FnCtxt`. I'm not sure a better name for it though ...

Reply:
The pattern comes from the `_match.rs` file in the directory above it. Seeing that, my assumption is that Rust would not allow keywords as identifiers (a reasonable assumption). 
Yes, I chose the name `_impl.rs` because of that and the file before I moved it was `fn_ctxt_impl.rs`.
I am fine with giving it a better name; I just cannot make one with the single brain I have.